packages/consensus/source/round-state.ts

```diff
@@ -224,13 +249,46 @@ export class RoundState implements Contracts.Consensus.IRoundState {
 		};
 	}
 
+	#getValidatorMajority(s: Map<string, { signature: string; blockId?: string }>): Map<string, { signature: string }> {
+		if (!this.hasMajorityPrevotes() || !this.hasMajorityPrecommits()) {
```

Comment:
Move it outside the method into aggregateMajorityPrevotes and aggregateMajorityPrecommits. Majority prevotes can be valid, without majority precommits.